lib/practice_exercises.rb

```diff
@@ -1,13 +1,55 @@
-
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(1)
 def remove_duplicates(list)
-  raise NotImplementedError, "Not implemented yet"
+  index_a = 0
+  index_b = 1
+  temp = 0
+  while index_b < list.length
+    if (list[index_a] == list[index_b])
+      temp = list[index_b]
+      list[index_b] = nil
+    end
+   
+    if (list[index_a] == nil) && (temp == list[index_b]) 
+      list[index_b] = nil
+    end
+
+    index_a += 1
+    index_b += 1
+
+  end
+  return list
+end
+
+# Time Complexity: O(mn) n = strings.length and m = smallest length in strings
+# Space Complexity: O(1)
+
+def all_same_index?(strings, index)
+  b = true
+  a = strings[0][index]
+  strings.each do |string|
+    if string[index] != a
+      b = false
+    end
+  end
+  return b
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
 def longest_prefix(strings)
```

Comment:
So this method in combination with your `all_same_index?` method would be O(m * n<sup>2</sup>) time complexity.